synex-frontend/src/codebase/indexer.ts

```
@@ -1,14 +1,397 @@
+
+  // Extract function names based on file extension
+  private extractFunctions(content: string, extension: string): string[] {
+    const functions: string[] = [];
+
+    switch (extension) {
+      case '.js':
+      case '.jsx':
+      case '.ts':
+      case '.tsx':
+        // JavaScript/TypeScript function patterns
+        const jsFunctionRegex = /(?:function\s+([a-zA-Z_$][a-zA-Z0-9_$]*)|([a-zA-Z_$][a-zA-Z0-9_$]*)\s*[:=]\s*(?:async\s+)?(?:function|\([^)]*\)\s*=>))/g;
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)